public/request/auth/update-password.php

```diff
@@ -15,17 +15,19 @@
 $newpass2 = requestInputPost('y');
 
 if ($newpass1 !== $newpass2) {
-    header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=passinequal");
-    exit;
-}
-
-if (mb_strlen($newpass1) < 8) {
-    header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=badnewpass");
+    if (!isset($passResetToken)) {
+        header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=passinequal");
+    } elseif (isValidPasswordResetToken($user, $passResetToken)) {
+        header("Location: " . getenv('APP_URL') . "/resetPassword.php?u=$user&t=$passResetToken&e=passinequal");
+    }
     exit;
 }
-
-if ($newpass1 == $user) {
-    header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=badnewpass");
+if (mb_strlen($newpass1) < 8 || $newpass1 == $user) {
+    if (!isset($passResetToken)) {
+        header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=badnewpass");
+    } elseif (isValidPasswordResetToken($user, $passResetToken)) {
+        header("Location: " . getenv('APP_URL') . "/resetPassword.php?u=$user&t=$passResetToken&e=badnewpass");
+    }
```

Comment:
same comment as above. you shouldn't call `exit` without doing a redirect first.